### DISCUSSION TITLE
Universal webview builder for desktop and mobile

### DISCUSSION THREAD

**Discussion by sebhernandezr** [tauri-apps/tauri#11359] [General]
Do you have plans to make the webview creation/manipulation universal for desktop and mobile? Thanks!

**Comment by FabianLars** [comment]
No real plans as of now. My guess would be that this would be part of multiwebview support but idk

Edit: Actually, i'm not even sure i 100% understood your question, could you expand it a bit more? Like, what issues you have with the current apis etc

**Reply by sebhernandezr** [reply]
I am creating my main app like you suggest for multiwebview: https://github.com/tauri-apps/tauri/blob/dev/examples/multiwebview/main.rs

Notice that I need to use `add_child`. However, `add_child` is only available on desktop: https://docs.rs/tauri/latest/tauri/window/struct.Window.html#method.add_child

I wanna have the same funtionatily for mobile.

**Reply by FabianLars** [reply]
Yeah okay, that's something we want to have at some point. No work has been done on that front yet though and i'm not aware of any concrete plans.

**Reply by sebhernandezr** [reply]
@FabianLars I wonder if there is any workaround I can do to have a multiwebview behavior in an android app. I need at least to be able to run an `index.html` from a react app in one half of the window and an external url in the other half.

**Reply by FabianLars** [reply]
if you don't need access to tauri's ipc then i'd imagine that doing it more or less the standard android way should work. I could swear someone on discord told me they did that but i can't find the message anymore 🤔 Anyway, i don't have any resources for that on hand that could help you.

**Reply by sebhernandezr** [reply]
I think it's this post: https://discord.com/channels/616186924390023171/1249696592177659965

**Reply by sebhernandezr** [reply]
![image](https://github.com/user-attachments/assets/9715c97e-6ded-48ce-9943-51e888f4b9c1)

That guy made a plugin for a recatcha thing, and he was trying to open a webview on android. I will take a look at his code at least how to run kotlin code, create the webview there and then put it in tauri.

**Comment by sebhernandezr** [comment]
@FabianLars I have one more question. I have managed to create a native webview in kotlin with the way Tauri recomend, by creating an [Android plugin](https://v2.tauri.app/develop/plugins/develop-mobile/#develop-an-android-plugin). However I wonder if I can acomplish the same thing with [ndk](https://github.com/rust-mobile/ndk) and [android-activity](https://github.com/rust-mobile/android-activity) crates.

**Reply by FabianLars** [reply]
no idea, sorry

**Comment by Ggcpp** [comment]
Any plan for this feature yet?

**Reply by FabianLars** [reply]
a bit of experimental work has been done for a customer but no ETA for its completion (or a more usable state) yet https://github.com/tauri-apps/tauri/compare/dev...feat/mobile-multi-window
